packages/react-components/src/organisms/Form.tsx

```diff
@@ -0,0 +1,76 @@
+import { ReactNode, useRef, useState, useEffect } from 'react';
+import { Prompt } from 'react-router-dom';
+import { css } from '@emotion/react';
+
+import { noop } from '../utils';
+import Toast from './Toast';
+
+const styles = css({
+  boxSizing: 'border-box',
+  height: '100%',
+  display: 'grid',
+  gridTemplateRows: 'max-content 1fr',
+  overflow: 'auto',
+});
+
+type FormProps = {
+  onSave?: () => void | Promise<void>;
+  validate?: () => boolean;
+  dirty: boolean; // mandatory so that it cannot be forgotten
+  children: (state: {
+    isSaving: boolean;
+    onSave: () => void | Promise<void>;
+  }) => ReactNode;
+};
+const Form: React.FC<FormProps> = ({
+  dirty,
+  children,
+  validate = () => true,
+  onSave = noop,
+}) => {
+  const formRef = useRef<HTMLFormElement>(null);
```

Comment:
What's the policy on using libraries, like e.g. formik for this?

Reply:
I intend to refactor that implementation to reuse this. I just wanted to get it tested first. Would appreciate any feedback.

Reply:
I am not sure I like the tripple wrap